### PR TITLE
Change currency to currency code

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | mentioned in #5173 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Component/Core/Model/Customer.php [hunk 1]
```diff
@@ -55,17 +55,17 @@ public function __construct()
     /**
      * {@inheritdoc}
      */
-    public function setCurrency($currency)
+    public function setCurrencyCode($currencyCode)
     {
-        $this->currency = $currency;
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Model/Customer.php]: `Assert::string` missing

File: src/Sylius/Bundle/WebBundle/Resources/views/Backend/Order/create.html.twig [hunk 2]
```diff
@@ -11,15 +11,15 @@
...
+    {#<h1><span class="glyphicon glyphicon-plus"></span> {{ 'sylius.ui.new_order'|trans|raw }}</h1>#}
+{#</div>#}
+{##}
+{#{{ form_errors(form) }}#}
+{##}
+{#<form action="{{ path('sylius_backend_order_create') }}" method="post" class="form-horizontal" novalidate>#}
+    {#<center>Not implemented yet.</center>#}
+    {#<hr>#}
+    {#{{ create() }}#}
+{#</form>#}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/WebBundle/Resources/views/Backend/Order/create.html.twig]: I believe this template and corresponding routing and references can be removed.

File: src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php [hunk 3]
```diff
@@ -77,9 +77,9 @@ protected function completeOrder(OrderInterface $order)
 
         $this->applyTransition(OrderCheckoutTransitions::TRANSITION_COMPLETE, $order);
         $this->get('sm.factory')->get($order, OrderTransitions::GRAPH)->apply(OrderTransitions::SYLIUS_CREATE, true);
-        if ($order->getCurrency() !== $currencyProvider->getBaseCurrency()) {
+        if ($order->getCurrencyCode() !== $currencyProvider->getBaseCurrency()) {
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php]: `$currencyProvider->getBaseCurrency()` returns `Currency` object, so using `!==` will always return true. IMO we can remove this `if` and left the logic to always save the exchange rate on `Order`.

**Comment by pjedrzejewski** [comment]
@GSadee Please add note to UPGRADE file.

**Comment by GSadee** [comment]
Done 👍

**Comment by michalmarcinkowski** [comment]
Good job Grzesiu! 👍

**Pull request merged**